### PR TITLE
Fixes #26468 - Index modules before errata

### PR DESCRIPTION
**Pull request by parthaa**
This commit indexes module streams before errata. While indexing errata we need to find the module streams to associate it.

### CODE REVIEW

**Comment by jlsherrill** [comment]
My steps for testing on master:
1) make sure this pr and https://github.com/Katello/katello/pull/8048 are checked out
2) create a repo and sync http://quartet.usersys.redhat.com/pub/repos/errata_test/modules_1/  (note this url isn't publicly accessible)
3) publish in a cv
4) change url on repository to http://quartet.usersys.redhat.com/pub/repos/errata_test/modules_2/  resync the repo
5) run:
```
hammer content-view version incremental-update --content-view-version-id=51 --errata-ids=RHEA-2013:0098
```
(change the content view id to the 1.0 version of your content view

6) examine the 1.1 content view version for the # of errata and modules, it should be one more of each than 1.0

My results:
image

File: lib/katello/repository_types/yum.rb [hunk 1]
```diff
@@ -3,8 +3,8 @@
   prevent_unneeded_metadata_publish
 
   content_type Katello::Rpm, :priority => 1, :pulp2_service_class => ::Katello::Pulp::Rpm, :user_removable => true
-  content_type Katello::Erratum, :priority => 2, :pulp2_service_class => ::Katello::Pulp::Erratum
-  content_type Katello::ModuleStream, :priority => 3, :pulp2_service_class => ::Katello::Pulp::ModuleStream
+  content_type Katello::ModuleStream, :priority => 2, :pulp2_service_class => ::Katello::Pulp::ModuleStream
+  content_type Katello::Erratum, :priority => 3, :pulp2_service_class => ::Katello::Pulp::Erratum
```
Comment (jlsherrill) [inline lib/katello/repository_types/yum.rb]: mind adding a test?

**Comment by bbuckingham** [comment]
@parthaa, As a user, I see the same behavior/output with this PR or using master.  Is this expected?

Following the steps from the https://projects.theforeman.org/issues/26468:
1) Sync https://partha.fedorapeople.org/test-repos/pteradactyl/ (assuming this repo has not been synced before. You might need to rake katello:reset before trying this otherwise )
2) Content -> Errata
3) Check the details for errata "RHEA-2012:0059" 
4) Go to Packages tab
Expected:
Module Stream Packages set for streams -> duck:0 and kangaroo:0
Actual:
No module streams

Results with and without the PR is the graphic below.

Packages for Errata  Duck_Kangaroo_Erratum - without fix

**Comment by omkarkhatavkar** [comment]
> havior/output with this PR or using master. Is this ex

@bbuckingham I think it looks good!

**Comment by omkarkhatavkar** [comment]
@parthaa @bbuckingham I even tested with hammer o/p. everything looks good as well.

**Comment by bbuckingham** [comment]
@omkarkhatavkar, Agreed.  The UI/page looks correct from what I see in the bug description.

That said, it was unclear to me if the fix is correct, since I see the same output in the UI before the fix and after.

**Comment by omkarkhatavkar** [comment]
@bbuckingham Yes, you are absolutely right. I even was not able to reproduce this issue on the second time. But I have seen this issue in between.

File: lib/katello/repository_types/yum.rb (hunk 1 above)
Comment (parthaa) [reply lib/katello/repository_types/yum.rb]: done

**Pull request merged**